crypto/cmp/cmp_ses.c

```diff
@@ -753,6 +761,39 @@ X509 *OSSL_CMP_exec_RR_ses(OSSL_CMP_CTX *ctx)
         goto err;
     }
 
+    /* check any present CertId in optional revCerts field */
+    if (rrep->certId != NULL) {
+        OSSL_CRMF_CERTID *cid;
+        OSSL_CRMF_CERTTEMPLATE *tmpl = 
+            sk_OSSL_CMP_REVDETAILS_value(rr->body->value.rr, rsid)->certDetails;
+        X509_NAME *issuer = OSSL_CRMF_CERTTEMPLATE_get0_issuer(tmpl);
+        ASN1_INTEGER *serial = OSSL_CRMF_CERTTEMPLATE_get0_serialNumber(tmpl);
+
+        if (sk_OSSL_CRMF_CERTID_num(rrep->certId) != num_RevDetails) {
+            CMPerr(CMP_F_OSSL_CMP_EXEC_RR_SES, CMP_R_WRONG_RP_COMPONENT_COUNT);
+            result = NULL;
+            goto err;
+        }
+        if ((cid = CMP_REVREPCONTENT_CertId_get(rrep, rsid)) == NULL) {
+            result = NULL;
+            goto err;
+        }
+        if (X509_NAME_cmp(issuer, OSSL_CRMF_CERTID_get0_issuer(cid)) != 0
+            || ASN1_INTEGER_cmp(serial,
+                                OSSL_CRMF_CERTID_get0_serialNumber(cid)) != 0) {
+            CMPerr(CMP_F_OSSL_CMP_EXEC_RR_SES, CMP_R_WRONG_CERTID_IN_RP);
+            result = NULL;
+            goto err;
+        }
+    }
+
+    /* check number of any optionally present crls */
+    if (rrep->crls != NULL && sk_X509_CRL_num(rrep->crls) != num_RevDetails) {
```

Comment:
Why is the number of CRLs here fixed to the number certificates which were requested to be revoked?
There could be several CRLs for the same requested certificate, when there are several [cRLIssuer](https://tools.ietf.org/html/rfc5280#section-4.2.1.13)s (not an every-day setup, but legally possible) - or there could be only one CRL for multiple revoked certificates (pretty common case once multiple revocations within one request will be supported).